src/components/RestaurantList.js

```diff
@@ -0,0 +1,73 @@
+import { RestaurantItem } from "./RestaurantItem.js";
+import { RestaurantFilter } from "./RestaurantFilter.js";
+import { loadRestaurants, saveRestaurants } from "../utils/storage.js";
+import DetailModal from "./DetailModal.js";
+
+export function RestaurantList(restaurantsData) {
+  let restaurants = loadRestaurants() || restaurantsData;
+
+  const state = { category: "전체", sort: "name" };
+
+  const $restaurantContainer = document.createElement("section");
+  $restaurantContainer.classList.add("restaurant-list-container");
+
+  const $filter = RestaurantFilter(state.category, state.sort);
+  const $restaurantList = document.createElement("ul");
+  $restaurantList.classList.add("restaurant-list");
+  $restaurantContainer.append($filter, $restaurantList);
+
+  const detailModal = new DetailModal();
```

Comment:
```suggestion
  const detailModal = new DetailModal({ delete: () => { ... } });
```
이렇게 delete 함수를 모달 내부로 넘겨줄 수 있으면 좋지 않을까요!?